[flang] Add support for the IARGC and GETARG legacy intrinsics (#196425)
llvm/llvm-project: flang/docs/Intrinsics.md

  call show_descriptor(a)
end subroutine test
```

### Non-Standard Intrinsics: IARGC

#### Description
`IARGC()` returns the number of arguments passed on the command line when the containing program was invoked.

#### Usage and Info
- **Standard:** GNU extension
- **Class:** Function
- **Syntax:** `RESULT = IARGC()`
- **Arguments:**
- **Return value:** The number of command line arguments, type `INTEGER(4)`

#### Example
```Fortran
program example_iargc
  integer :: n
  n = iargc()
  print *, "Argument count:", n
end program
```

### Non-Standard Intrinsics: GETARG

#### Description
`GETARG(POS, VALUE)` retrieves the `POS`-th argument that was passed on the command line when the containing program was invoked.
After `GETARG` returns, the `VALUE` argument holds the `POS`-th command line argument.

> If `VALUE` cannot hold the argument, it is truncated to fit the length of `VALUE`.

Reviewer: ```suggestion
If `VALUE` cannot hold the argument, the argument is truncated to fit in `VALUE`.
```
